byterun/weak.c

```diff
@@ -285,7 +285,8 @@ CAMLprim value caml_ephe_get_key_copy (value ar, value n)
 
   if (is_ephe_key_none(ar, offset)) CAMLreturn (None_val);
   v = Field (ar, offset);
-  if (Is_block (v) && Is_in_heap_or_young(v)) {
+  /** Don't copy custom_block #7279 */
+  if (Is_block (v) && Is_in_heap_or_young(v) && Tag_val(v) != Custom_tag ) {
     elt = caml_alloc (Wosize_val (v), Tag_val (v));
```

Comment:
What happens here if `Tag_val(v) = Forward_tag`? I am thinking that this call to `caml_alloc` might short-circuit `v` and make it a custom block (of  a different size than what's allocated for `elt`).